libra-node/src/main_node.rs

```diff
@@ -160,19 +160,20 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
                 if consensus_network_handles.is_some() {
                     panic!("There can be at most one validator network!");
                 }
-
-                // Set up to listen for network configuration changes from StateSync.
-                // TODO:  move this inside network_builder.
-                if let Some(conn_mgr_reqs_tx) = network_builder.conn_mgr_reqs_tx() {
+                // HACK: gossip relies on on-chain discovery for the eligible peers update.
+                if network_config.discovery_method != DiscoveryMethod::Onchain {
+                    let conn_mgr_reqs_tx = network_builder
+                        .conn_mgr_reqs_tx()
+                        .expect("ConnectivityManager must be installed for on-chain discovery");
```

Comment:
Maybe a more appropriate comment would be:
Gossip protocol requires p2p networks which needs ConnectivityManager.  

`expect("ConnectivityManager is necessary for non-on-chain discovery")`